assets/page/SettingsAccount.svelte

```diff
@@ -9,19 +9,36 @@ import {getContext} from 'svelte';
 import {l} from '../js/i18n';
 import {route} from '../store/Route';
 
-const colorSchemeOptions = [['auto', 'Auto'], ['dark', 'Dark'], ['light', 'Light']];
 const user = getContext('user');
 const updateUserOp = user.api.operation('updateUser');
 
+// "colorSchemeOptions" will get calculated each time "theme" changes
+$: colorSchemeOptions = calculateColorSchemeOptions(theme);
+$: colorScheme = resetColorScheme(theme);
+
+function calculateColorSchemeOptions(id) {
+  const options = [['auto', 'Auto']];
+  const theme = user.themes[id] || {variants: {}};
+  if (theme.variants.dark) options.push(['dark', 'Dark']);
+  if (theme.variants.light) options.push(['light', 'Light']);
+  return options;
+}
+
+function resetColorScheme(id) {
+  colorScheme = 'auto';
+  return colorScheme;
+}
+
 const themes = Object.keys(user.themes).map(id => {
   let name = user.themes[id].name;
   const colorSchemes = Object.keys(user.themes[id].variants || {}).filter(cs => cs != 'default');
   if (colorSchemes.length) name += ' (' + colorSchemes.sort().join('/') + ')';
   return [id, name];
 });
 
+
 let formEl;
-let colorScheme = user.colorScheme;
+colorScheme = user.colorScheme;
```

Comment:
With the suggested `calculateColorSchemeOptions()` change above, then this must be changed back to `let ...`, which is the right thing to do, since it is bound to the SelectField component.

Reply:
Done